experimenter/experimenter/features/__init__.py

```diff
@@ -80,30 +82,53 @@ def get_jsonschema(self):
 
 
 class Features:
-    _features = None
+    _features: Optional[list[Feature]] = None
+
+    @classmethod
+    def _read_manifest(
+        cls,
+        application: ApplicationConfig,
+        manifest_path: Path,
+        version: Version = None,
+    ):
+        with manifest_path.open() as manifest_file:
+            application_data = yaml.safe_load(manifest_file)
+            manifest = FeatureManifest.parse_obj(application_data)
+
+            for feature_slug, feature_model in manifest.__root__.items():
+                yield Feature(
+                    slug=feature_slug,
+                    application_slug=application.slug,
+                    model=feature_model.__root__,
+                    version=version,
+                )
 
     @classmethod
     def _load_features(cls):
         features = []
+        version_re = re.compile(r"^v(?P<major>\d+)\.(?P<minor>\d+)\.(?P<patch>\d+)")
 
         for application in NimbusConstants.APPLICATION_CONFIGS.values():
-            application_yaml_path = os.path.join(
-                settings.FEATURE_MANIFESTS_PATH, application.slug, "experimenter.yaml"
-            )
+            application_dir: Path = settings.FEATURE_MANIFESTS_PATH / application.slug
+            application_yaml_path = application_dir / "experimenter.yaml"
```

Comment:
```suggestion
            application_yaml_path = Path(application_dir, "experimenter.yaml")
```

Reply:
using / is idiomatic when joining `Path`s.